tests/contracts/test_deposit.py

```diff
@@ -25,37 +43,76 @@ def test_deposit(registration_contract, a0, w3, success, amount_deposit, assert_
         )
 
 
-def test_no_reuse_of_pubkey(registration_contract, a0, w3, assert_tx_failed):
+def test_deposit_log(registration_contract, a0, w3):
```

Comment:
* should parametrize with 2 or 3 different deposit amounts.
* should run a test that issues two deposits to ensure that `previous_receipt_root` field becomes non-zero and `deposit_count` increments